data/core/commandview.lua

```diff
@@ -274,18 +275,31 @@ function CommandView:get_suggestion_line_height()
 end
 
 
-function CommandView:update_suggestions()
-  local t = self.state.suggest(self:get_text()) or {}
+function CommandView:update_suggestions(preserve)
+  local text = self:get_text()
+  local t = self.state.suggest(text == self.save_suggestion and "" or text) or {}
   local res = {}
   for i, item in ipairs(t) do
     if type(item) == "string" then
       item = { text = item }
     end
     res[i] = item
   end
+  if preserve and self.suggestions and self.suggestion_idx ~= 1 then
+    local current_suggestion_idx
```

Comment:
Nit, maybe `next_suggestion_idx` or even `future...`.

Reply:
I mean, `current_suggestion_idx`; i.e. it's looking to see where we currently are in the new list of suggestions. Maybe `new_suggestion_idx`? Iunno.

Reply:
Yeah, `new...` sounds better.